build-packages/scandipwa-development-toolkit-core/src/extend/js-generation.ts

```diff
@@ -0,0 +1,211 @@
+import { ResourceType } from '../types';
+
+import {
+    FileInformation,
+    ExportType,
+    StylesOption,
+    ExportData,
+} from '../types/extend-component.types';
+import { getImportPath } from '../util/js-generation';
+import { capitalize, decapitalize } from '../util/misc';
+
+import { getStyleFileName } from './scss-generation';
+
+const isMapping = (name: string) => ['mapStateToProps', 'mapDispatchToProps'].includes(name);
+
+const getPrefixedName = (name: string, moduleAlias: string) => {
+    const isCapitalized = (word: string) => word[0].toUpperCase() === word[0];
+    const isUpperCase = (word: string) => word.toUpperCase() === word;
+
+    if (isUpperCase(name)) {
+        return `${moduleAlias.toUpperCase()}_${name}`;
+    }
+
+    if (isCapitalized(name)) {
+        return [moduleAlias, name].map(capitalize).join('');
+    }
+
+    return [decapitalize(moduleAlias), capitalize(name)].join('');
+}
+
+const generateAdditionalImportString = (originalCode: string, defaultExportCode?: string): string => {
+    if (!defaultExportCode) { 
+        return ''; 
+    }
+
+    const exdfWords = [
+        ...new Set(
+            defaultExportCode.match(/\w+\(/gm)?.filter(
+                word => !['export', 'default'].includes(word)
+            ) || []
+        )
+    ].map(
+        // Cut away parentheses
+        str => str.slice(0, str.length - 1)
+    );
+
+    return (exdfWords).reduce(
+        (acc, importableName): Array<string | undefined> => {
+            const library = originalCode.match(
+                new RegExp(`import.+${importableName}.+from '(?<library>.+)'`)
+            )?.groups?.library;
+
+            const braces = !!originalCode.match(
+                new RegExp(`import.+{.*${importableName}.*}.+from '(?<library>.+)'`)
+            );
+
+            if (!library) {
+                return acc;
+            }
+
+            acc.push(
+                ''.concat('import ')
+                .concat(braces ? '{ ' : '')
+                .concat(`${importableName}`)
+                .concat(braces ? ' }' : '')
+                .concat(` from '${library}';`)
```

Comment:
A monster! But I do not know the better way :)

Reply:
Expect the:
```js
[
  'import ',
  braces ? '{ ' : '',
  importableName,
  braces ? ' }' : '',
  ` from '${library}';`
].join('')

```

Reply:
Or even better:
```js
[
  'import',
  braces ? '{' : '',
  importableName,
  braces ? '}' : '',
  `from '${library}';`
].join(' ')
```

Reply:
Replaced concats with joins where it is appropriate.